content/en/guide/v10/typescript.md

```diff
@@ -56,69 +57,53 @@ If you use TypeScript within a Babel toolchain, set `jsx` to `preserve` and let
 
 Rename your `.jsx` files to `.tsx` for TypeScript to correctly parse your JSX.
 
-## Typing components
+## Typing Components
+
+### Functional Components
 
-There are different ways to type components in Preact. Class components have generic type variables to ensure type safety. TypeScript sees a function as functional component as long as it returns JSX. There are multiple solutions to define props for functional components.
+The type corresponding to functional components is `FunctionComponent` type. This is type definition of `FunctionComponent` type:
+
+```ts
+interface FunctionComponent<P = {}> {
+  (props: RenderableProps<P>, context?: any): VNode<any> | null;
+  displayName?: string;
+  defaultProps?: Partial<P>;
+}
+```
 
-### Function components
+> 💁 FunctionalComponent type is alias of FunctionComponent type.
 
-Typing regular function components is as easy as adding type information to the function arguments.
+You can implement a functional component in TypeScript, as below.
 
 ```tsx
-type MyComponentProps = {
+type Props = {
   name: string;
   age: number;
 };
 
-function MyComponent({ name, age }: MyComponentProps) {
+const MyComponent: FunctionComponent<Props> = function ({ name, age }) {
   return (
     <div>
-      My name is {name}, I am {age.toString()} years old.
+      My name is {name}, I am {age} years old.
     </div>
   );
 }
 ```
 
-You can set default props by setting a default value in the function signature.
-
-```tsx
-type GreetingProps = {
-  name?: string; // name is optional!
-}
+`children` is `ComponentChildren` type. `children` as `ComponentChildren` type is added to props in `RenderableProps` type:
 
-function Greeting({ name = "User" }: GreetingProps) {
-  // name is at least "User"
-  return <div>Hello {name}!</div>
-}
+```ts
+type RenderableProps<P, RefType = any> = P & Readonly<Attributes & { children?: ComponentChildren; ref?: Ref<RefType> }>;
 ```
 
-Preact also ships a `FunctionComponent` type to annotate anonymous functions. `FunctionComponent` also adds a type for `children`:
+You do not need to add `children` as `ComponentChildren` type to `Props` type:
 
 ```tsx
-import { h, FunctionComponent } from "preact";
-
-const Card: FunctionComponent<{ title: string }> = ({ title, children }) => {
-  return (
-    <div class="card">
-      <h1>{title}</h1>
-      {children}
-    </div>
-  );
-};
-```
-
-`children` is of type `ComponentChildren`. You can specify children on your own using this type:
-
-
-```tsx
-import { h, ComponentChildren } from "preact";
-
-type ChildrenProps = {
+type Props = {
   title: string;
-  children: ComponentChildren;
```

Comment:
Since `children` as `ComponentChildren` type is added to props in `RenderableProps` type, this line is unnecessary.